equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,420 @@
+	}
+	updateFg := v4.Network{}
+	updateFg, err = waitForFabricNetworkUpdateCompletion(d.Id(), meta, ctx)
+
+	if err != nil {
+		return diag.Errorf("errored while waiting for successful Fabric Network update, response %v, error %v", res, err)
+	}
+
+	d.SetId(updateFg.Uuid)
+	return setFabricNetworkMap(d, updateFg)
+}
+
+func waitForFabricNetworkUpdateCompletion(uuid string, meta interface{}, ctx context.Context) (v4.Network, error) {
+	log.Printf("Waiting for Network update to complete, uuid %s", uuid)
+	stateConf := &resource.StateChangeConf{
+		Target: []string{string(v4.PROVISIONED_NetworkEquinixStatus)},
+		Refresh: func() (interface{}, string, error) {
+			client := meta.(*config.Config).FabricClient
+			dbConn, _, err := client.NetworksApi.GetNetworkByUuid(ctx, uuid)
+			if err != nil {
+				return "", "", equinix_errors.FormatFabricError(err)
+			}
+			return dbConn, string(*dbConn.Operation.EquinixStatus), nil
+		},
+		Timeout:    2 * time.Minute,
+		Delay:      30 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	inter, err := stateConf.WaitForStateContext(ctx)
+	dbConn := v4.Network{}
+
+	if err == nil {
+		dbConn = inter.(v4.Network)
+	}
+	return dbConn, err
+}
+
+func waitUntilFabricNetworkIsProvisioned(uuid string, meta interface{}, ctx context.Context) (v4.Network, error) {
+	log.Printf("Waiting for Fabric Network to be provisioned, uuid %s", uuid)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			string(v4.PROVISIONING_NetworkEquinixStatus),
+		},
+		Target: []string{
+			string(v4.PROVISIONED_NetworkEquinixStatus),
+		},
+		Refresh: func() (interface{}, string, error) {
+			client := meta.(*config.Config).FabricClient
+			dbConn, _, err := client.NetworksApi.GetNetworkByUuid(ctx, uuid)
+			if err != nil {
+				return "", "", equinix_errors.FormatFabricError(err)
+			}
+			return dbConn, string(*dbConn.Operation.EquinixStatus), nil
+		},
+		Timeout:    5 * time.Minute,
+		Delay:      30 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	inter, err := stateConf.WaitForStateContext(ctx)
+	dbConn := v4.Network{}
+
+	if err == nil {
+		dbConn = inter.(v4.Network)
+	}
+	return dbConn, err
+}
+
+func resourceFabricNetworkDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	_, resp, err := client.NetworksApi.DeleteNetworkByUuid(ctx, d.Id())
+	if err != nil {
+		errors, ok := err.(v4.GenericSwaggerError).Model().([]v4.ModelError)
+		if ok {
+			// EQ-3040055 = There is an existing update in REQUESTED state
+			if equinix_errors.HasModelErrorCode(errors, "EQ-3040055") {
+				return diags
+			}
+		}
+		return diag.Errorf("error response for the Fabric Network delete. Error %v and response %v", equinix_errors.FormatFabricError(err), resp)
```

Comment:
This line should just be `return diag.FromErr(equinix_errors.FormatFabricError(err))`